hypershift-operator/controllers/nodepool/azure.go

```diff
@@ -33,11 +33,14 @@ func azureMachineTemplateSpec(hcluster *hyperv1.HostedCluster, nodePool *hyperv1
 		return nil, fmt.Errorf("failed to determine subnet name for Azure machine: %w", err)
 	}
 
+	if nodePool.Spec.Platform.Azure.Image.ImageID == "" && nodePool.Spec.Platform.Azure.Image.MarketplaceImageInfo == nil {
```

Comment:
This should never happen by design with the cel validation you put in place right? can we add a // comment stating that?
Also can the error indicate user action, e.g. "either ImageID or MarketplaceImageInfo need to be filled"